azure-east/arc-system/application-runner-scale-set-dind.yaml

```diff
@@ -0,0 +1,92 @@
+metadata:
+  name: arc-runners-d-in-d  # Name of the Argo CD application
+  namespace: argocd  # Namespace where Argo CD application resides
+  finalizers:
+    - resources-finalizer.argocd.argoproj.io  # Ensures resources are cleaned up when the application is deleted
+spec:
+  project: arc  # Project in Argo CD where the application belongs
+  source:
+    repoURL: ghcr.io/actions/actions-runner-controller-charts  # Repository URL for the Helm chart
+    targetRevision: 0.6.1  # Version of the Helm chart to use
+    chart: gha-runner-scale-set  # Helm chart name
+    helm:
+      releaseName: "arc-runners-d-in-d"  # Override the Helm release name
+      values: |  # Inline Helm values for customizing the deployment
+        githubConfigUrl: "https://github.com/containerization-vgroup"  # URL for the GitHub configuration
+        githubConfigSecret: "pre-defined-secret"  # Kubernetes secret with GitHub token
+        controllerServiceAccount:
+          namespace: arc-system  # Namespace for the controller's service account
+          name: arc-gha-rs-controller  # Name of the service account
+        minRunners: 2  # Minimum number of runners
+        maxRunners: 10  # Maximum number of runners
+        runnerGroup: "arc-runners-d-in-d"  # GitHub runner group name
+        listenerTemplate:
+          spec:
+            containers:
+            - name: listener  # Name of the listener container
+          metadata:
+            annotations:
+              "prometheus.io/scrape": "true"  # Enable Prometheus scraping
+              "prometheus.io/path": "/metrics"  # Metrics path for Prometheus
+              "prometheus.io/port": "8080"  # Port for Prometheus to scrape
+        template:
+          spec:
+            containers:
+              - name: runner  # Name of the runner container
+                image: ghcr.io/actions/actions-runner:latest  # Docker image for the runner
+                command: ["/home/runner/run.sh"]  # Command to run inside the container
+                env:
+                - name: runtime
+                  value: "kubernetes"  # Set runtime environment
+                - name: DOCKER_HOST
+                  value: "tcp://localhost:2375"  # Set Docker host for DIND
+                volumeMounts:
+                - name: work
+                  mountPath: /home/runner/work  # Mount path for work directory
+                - name: docker-socket
+                  mountPath: /var/run/docker.sock  # Mount path for Docker socket
+            volumes:
+            - name: work
+              emptyDir: {}  # Create an empty directory volume for work
+            - name: docker-socket
+              emptyDir: {}  # Create an empty directory volume for Docker socket (supports rootless mode)
+            initContainers:
+            - name: dind-rootless  # Name of the init container for DIND rootless
+              image: docker:20.10-dind-rootless  # Docker image for DIND rootless
```

Comment:
If you run into issues with this, here are some potential areas to look at
- You may need `restartPolicy: always` on init container ([docs link](https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/#pod-sidecar-containers)), assuming this runs a daemon. Init containers generally must exist before anything in the "containers" section starts, having the restart policy lets you start the init container before other containers without it needing to exit
- consider using a newer version of the rootless container, `26.1` looks like the latest https://hub.docker.com/_/docker
- just a not from the [Docker docs](https://docs.docker.com/engine/security/rootless/#rootless-docker-in-docker) - "The `docker:<version>-dind-rootless` image runs as a non-root user (UID 1000). However, `--privileged` is required for disabling seccomp, AppArmor, and mount masks." - may need to set additional security contexts or set privileged if there are failures with this